scanner/src/test/java/ca/on/oicr/gsi/runscanner/scanner/processor/AbstractProcessorTest.java

```diff
@@ -29,7 +31,24 @@ protected final void checkDirectory(String root)
           null); // We delete this because it is going to change during updates to dependencies.
       result.setSequencerFolderPath(
           null); // We delete this because it is going to be different in each environment.
-      result.setMetrics(null); // We delete these because changes in metrics are non-critical.
+
+      // For Illumina processors only
+      if (clazz.equals(IlluminaNotificationDto.class)) {
+        // Load the metrics into a Jackson JsonNode to test for equality
+        ObjectMapper MapperTest = new ObjectMapper();
+        JsonNode jsonNodeResult = MapperTest.readTree(result.getMetrics());
+        JsonNode jsonNodeReference = MapperTest.readTree(reference.getMetrics());
+        if (!jsonNodeReference.equals(jsonNodeResult)) {
+          assertEquals(reference, result);
+        }
+        // We know metrics equal, set reference to null to match result metrics
+        reference.setMetrics(null);
```

Comment:
This is currently not really doing anything since you removed metrics comparison from `NotificationDto#equals`. Does the String version of metrics match? If yes, don't remove from either (`reference` or `result`). If no, remove from both.